src/firefly/java/edu/caltech/ipac/firefly/server/visualize/FitsCacher.java

```diff
@@ -59,9 +64,13 @@
                     else {
                         Fits fits= null;
                         try {
+                            if (memCache != null) {
+                                memCache.put(key, new ObjectSizeEngineWrapper.BluffSize(fitsFile.length()));
+                                memCache.put(key, null);
```

Comment:
I am not quite understand what it is doing here? Why the same key was put twice with different value?